pkg/client/add_events_test.go

```diff
@@ -41,6 +41,8 @@ import (
 
 const RetryBase = time.Second
 
+var attempt = atomic.Int32{}
```

Comment:
Any reason why you made this a module level global?

Won't this cause potential issue with cross test pollution? (aka each test will need to reset it to ensure it starts with the clean environment / 0 value).

I guess you have done it since you also moved mock server function outside. Which could be fine, but yeah, we need to make sure each test function starts with a "clean environment" and there is no cross test pollution / state sharing going on.

Reply:
Looks like default test runner doesn't support that and you need to implement it yourself - https://medium.com/nerd-for-tech/setup-and-teardown-unit-test-in-go-bd6fa1b785cd

And even with using setUp and tearDown like functionality, there could be issues in case multiple tests in the same file run concurrently using go routines.

So if everything is passing right now, we could leave it as-is, but we should keep an eye on it in case we start seeing weird occasional failures / races in the future.